scripts/install_third_party_libs_test.py

```diff
@@ -44,41 +45,122 @@
+            'url_retrieve_is_called': False,
+            'recursive_chmod_is_called': False
+        }
+        self.check_main_function_calls = {
+            'ensure_pip_library_is_installed_is_called': False,
+            'install_third_party_main_is_called': False,
+            'setup_main_is_called': False,
+            'setup_gae_main_is_called': False,
+            'pre_commit_hook_main_is_called': False
+        }
+        self.expected_check_main_function_calls = {
+            'ensure_pip_library_is_installed_is_called': True,
+            'install_third_party_main_is_called': True,
+            'setup_main_is_called': True,
+            'setup_gae_main_is_called': True,
+            'pre_commit_hook_main_is_called': True
         }
-        self.print_arr = []
-        def mock_check_call(unused_cmd_tokens, *args, **kwargs):  # pylint: disable=unused-argument
+        def mock_check_call(args: List[str], **kwargs: Union[int, None]) -> Ret:  # type: ignore[name-defined] # pylint: disable=unused-argument,undefined-variable
             self.check_function_calls['check_call_is_called'] = True
             class Ret:
                 """Return object with required attributes."""
 
-                def __init__(self):
+                def __init__(self) -> None:
                     self.returncode = 0
-                def communicate(self):
+                def communicate(self) -> Tuple[bytes, bytes]:
                     """Return required method."""
                     return b'', b''
             return Ret()
-        def mock_check_call_error(*args, **kwargs) -> None:  # pylint: disable=unused-argument
+        def mock_check_call_error(
+                args: List[str], **kwargs: Union[int, None]) -> None: # pylint: disable=unused-argument
             """Raise the Exception resulting from a failed check_call()"""
             self.check_function_calls['check_call_is_called'] = True
             raise subprocess.CalledProcessError(-1, args[0])
-        def mock_popen_error_call(unused_cmd_tokens, *args, **kwargs):  # pylint: disable=unused-argument
+        def mock_popen_error_call(
+            args: List[str], **kwargs: Union[int, None]) -> Ret:  # type: ignore[name-defined] # pylint: disable=unused-argument,undefined-variable
             class Ret:
                 """Return object that gives user-prefix error."""
 
-                def __init__(self):
+                def __init__(self) -> None:
                     self.returncode = 1
-                def communicate(self):
+                def communicate(self) -> Tuple[bytes, bytes]:
                     """Return user-prefix error as stderr."""
                     return b'', b'can\'t combine user with prefix'
             return Ret()
-        def mock_print(msg, end=''):  # pylint: disable=unused-argument
-            self.print_arr.append(msg)
+        def mock_ensure_directory_exists(path: str) -> None:  # pylint: disable=unused-argument
```

Comment:
I think we can avoid this pylint disable statement by either using `_` or using an argument name that starts with `_`. WDYT?

ditto elsewhere.

Reply:
Done.